Firestore/Swift/CHANGELOG.md

```diff
@@ -1,9 +1,7 @@
 # 10.0.0
-- [changed] **Breaking Change:** Made the `@DocumentID` property wrapper value
-  setter internal to clarify that the value is ignored during writes. (#9368)
-- [changed] Initializing a `@DocumentID` property wrapper with a non-nil value
-  or using the `@DocumentID` property wrapper value setter will log a warning.
-  This is because the set value will be ignored. (#9368)
+- [changed] Passing a non-nil value to the `@DocumentID` property wrapper's
+  constructor or setter will log a warning and the set value will be ignored.
+  (#9368)
```

Comment:
Isn't the breaking change that we removed the public visibility from this init:

<img width="744" alt="Screen Shot 2022-10-05 at 1 00 37 PM" src="https://user-images.githubusercontent.com/36927374/194118778-750b3805-06e8-46d2-9599-8d9097f4adbf.png">

_(Diff from #10167)_

Reply:
I'd argue that this isn't a breaking change since it was only visible when `DocumentID` _wasn't_ used as a property wrapper, i.e., `DocumentID<DocumentReference>(from: docRef)`.

The following is no longer possible (though I can't think of a use case):
```
let userDoc: DocumentReference = Firestore.firestore().document("users/foo")
do {
  let docID = try DocumentID<DocumentReference>(from: userDoc)
  if let docRef = docID.wrappedValue {
    print("Document ID: \(docRef.documentID)")
  }
} catch {
  print("DocumentID Error: \(error)")
}
```

The visibility change doesn't affect usage of at `@DocumentID` though. The following is still possible (though it will trigger our warnings at runtime:
```
struct User: Codable {
  @DocumentID var docRef: DocumentReference? = Firestore.firestore().document("users/foo")
}
```

We could list the above as a breaking change but I think it could cause confusion since DocumentID has always been intended to be used as a property wrapper. Let me know if you feel differently, of course, though.

Reply:
Thanks for the thorough answer! I think that, if something is no longer possible (even if it doesn't have a use case), it's still a breaking change. Curious what @morganchen12 and @paulb777  think though.

Reply:
I've seen plenty of examples of people using tooling in completely unexpected ways, so that would be an argument to call it breaking. On the flip side, if we're convinced it won't actually break anything, I'm fine to remove the breaking notification. I'll defer to Andrew's judgement since he's looked at it the most, unless Morgan feels strongly one way or the other.

Reply:
Let's err on the side of caution. I reworded the changelog and specifically mentioned that `@DocumentID` is unchanged to avoid any confusion.